File: src/server/Adfnet.Core/ValueObjects/LeafMenu.cs
using System;

namespace Adfnet.Core.ValueObjects
{
    public class LeafMenu
    {
        public Guid Id { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Address { get; set; }
        public string Icon { get; set; }
        public string Description { get; set; }
        public virtual ChildMenu Parent { get; set; }
    }
}

File: src/client/web/private/src/app/models/leaf-menu.ts
import { ChildMenu } from './child-menu';

export class LeafMenu {
    id: string;
    code: string;
    name: string;
    address: string;
    icon: string;
    description: string;
    parent: ChildMenu;

    constructor() {
        this.parent = new ChildMenu();
    }
}
